### PR TITLE
Rewrite write_to_png() for QR class

### PR DESCRIPTION
**Pull request by TheXCoders**
A new function that can convert a QR Code object to a PNG file.

### CODE REVIEW

**Comment by jbakosi** [comment]
@TheXCoders please merge with main before this can be reviewed. Thanks.

**Review by layters — APPROVED** [review]

**Comment by TheXCoders** [comment]
In order to make things more clean, I need to make a new PR.

File: src/qr.cpp [hunk 1]
```diff
@@ -2,11 +2,10 @@
 
 neroshop::QR::QR() {}
 ////////////////////
-neroshop::QR::QR(std::string filename, int image_size, int min_module_pixel_size, std::string text,
+neroshop::QR::QR(std::string filename, int image_size, std::string text,
```
Comment (jbakosi) [inline src/qr.cpp:5]: Is there a reason to make a copy of the input argument strings, `filename` and `text`? If not, I would avoid them with
```cpp
neroshop::QR::QR( const std::string& filename, int image_size, const std::string& text, bool overwrite, qrcodegen::QrCode::Ecc ecc)
```

Comment (TheXCoders) [reply src/qr.cpp:5]: You're right, I made the modification on the new PR.

**Pull request closed without merging**